libndt7/src/main/java/io/ooni/mk/nuvolari/libndt7/Client.java

```diff
@@ -1,100 +1,155 @@
 package io.ooni.mk.nuvolari.libndt7;
 
+import android.support.annotation.CheckResult;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.security.cert.X509Certificate;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.TimeUnit;
 
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
 import okhttp3.OkHttpClient;
 import okhttp3.Request;
-import okhttp3.WebSocketListener;
-import okhttp3.WebSocket;
 import okhttp3.Response;
+import okhttp3.WebSocket;
+import okhttp3.WebSocketListener;
 import okio.ByteString;
 
-import org.json.JSONException;
-import org.json.JSONObject;
-
 // TODO(bassosimone): do we need locking for this class?
 
 public class Client extends WebSocketListener {
-    public Client(Settings settings) {
+
+    @NonNull
+    private final Settings settings;
+
+    private double count = 0.0;
+    private boolean rv = true;
+    private long t0 = 0;
+    private long tLast = 0;
+    private final long measurementInterval = TimeUnit.NANOSECONDS.convert(250, TimeUnit.MILLISECONDS);
+
+    @NonNull
+    private static final String TAG = "Client";
+
+    public Client(@NonNull Settings settings) {
         this.settings = settings;
     }
 
-    public void onLogInfo(String message) {
+    public void onLogInfo(@Nullable String message) {
+        Log.d(TAG, "onLogInfo: " + message);
         // NOTHING
     }
 
-    public void onError(String error) {
+    public void onError(@Nullable String error) {
+        Log.d(TAG, "onError: " + error);
         // NOTHING
     }
 
-    public void onServerDownloadMeasurement(Measurement measurement) {
+    public void onServerDownloadMeasurement(@NonNull Measurement measurement) {
+        Log.d(TAG, "onServerDownloadMeasurement: " + measurement);
         // NOTHING
     }
 
-    public void onClientDownloadMeasurement(Measurement measurement){
+    public void onClientDownloadMeasurement(@NonNull Measurement measurement){
+        Log.d(TAG, "onClientDownloadMeasurement: " + measurement);
         // NOTHING
     }
 
     @Override
     public final void onOpen(WebSocket ws, Response resp) {
-        onLogInfo("WebSocket connection established");
+        onLogInfo("WebSocket onOpen");
     }
 
     @Override
-    public final void onMessage(WebSocket ws, String text) {
+    public final void onMessage(WebSocket ws,
+                                String text) {
+        onLogInfo("WebSocket onMessage");
         count += text.length();
         periodic();
 
         Measurement measurement = new Measurement();
-        JSONObject doc;
+
+        @NonNull
+        final JSONObject doc;
```

Comment:
`final` here is to avoid races?

Reply:
Just good practice to help the compiler insure that we initialized correctly